alea/submitters/local.py

```diff
@@ -187,15 +191,22 @@ def submit(
                 )
 
             # update poi according to poi_expectation
-            runner_args["statistical_model_args"].pop("limit_threshold", None)
-            runner = Runner(**runner_args)
-            expectation_values = runner.model.get_expectation_values(
-                **{**nominal_values, **generate_values}
-            )
-            # in some rare cases the poi is not a rate multiplier
-            # then the poi_expectation is not in the nominal_expectation_values
-            component = self.poi.replace("_rate_multiplier", "")
-            poi_expectation = expectation_values.get(component, None)
+            if runner.input_poi_expectation:
+                poi_expectation = generate_values.get("poi_expectation")
+                # nominal_values are passed to update_poi to update the poi
+                # like wimp_mass, livetime, etc.
+                generate_values = runner.update_poi(
+                    runner.model, self.poi, generate_values, nominal_values
```

Comment:
I think this won't work with static parameters. That was the entire point of the previous PR

Reply:
I ran it quickly and yes, this crashes if you have static parameters (as it should).

Reply:
The recent commit 7b4f38fe8c4bbb9d1dfb41ba95638deaa3a32040 makes the PR compatible with static parameters.